runtime/src/main/java/org/corfudb/runtime/view/replication/ChainReplicationProtocol.java

```diff
@@ -0,0 +1,185 @@
+package org.corfudb.runtime.view.replication;
+
+import io.netty.buffer.ByteBuf;
+import lombok.extern.slf4j.Slf4j;
+import org.corfudb.protocols.logprotocol.StreamData;
+import org.corfudb.protocols.logprotocol.StreamedLogData;
+import org.corfudb.protocols.wireprotocol.ILogData;
+import org.corfudb.runtime.exceptions.OverwriteException;
+import org.corfudb.runtime.exceptions.RecoveryException;
+import org.corfudb.runtime.view.Layout;
+import org.corfudb.util.AutoClosableByteBuf;
+import org.corfudb.util.CFUtils;
+import org.corfudb.util.serializer.Serializers;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.Map;
+import java.util.UUID;
+
+/**
+ * Created by mwei on 4/6/17.
+ */
+@Slf4j
+public class ChainReplicationProtocol extends AbstractReplicationProtocol {
+
+    final Layout layout;
+
+    public ChainReplicationProtocol(IHoleFillPolicy holeFillPolicy, Layout layout) {
+        super(holeFillPolicy);
+        this.layout = layout;
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public void write(long globalAddress,
+                      @Nonnull Map<UUID, StreamData> entryMap) throws OverwriteException {
+        int numUnits = layout.getSegmentLength(globalAddress);
+
+        try (AutoClosableByteBuf b = new AutoClosableByteBuf()) {
```

Comment:
To be sure to release the buffer using try-with-resources. Internally autoclosable bytebuf uses a unpooled allocator but it can use a pooled one in the future.